### PR TITLE
Changed default ciphers

### LINKED ISSUE #50: Custom Cipher suites
Allow end-users to specify the cipher suites they want HAProxy to use, with a sane default, probably something like:

```
 ssl-default-bind-ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA
```

### PR DESCRIPTION
**Pull request by lcacciagioni**
With this commit we can improve the default ciphers to be more safe than what they are today. And maybe is possible to close #50.

### CODE REVIEW

File: jobs/haproxy/spec [hunk 1]
```diff
@@ -49,7 +49,7 @@ properties:
...
+    default: ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS
```
Comment (geofffranks) [inline jobs/haproxy/spec]: @lcacciagioni Can we not include EDH-RSA-DES-CBC3-SHA, ECDHE-RSA-DES-CBC3-SHA, and DES-CBC3-SHA in here? All three are listed as week on ssllabs tests, and the first two were disabled previously.

**Comment by lcacciagioni** [comment]
Id possible to disable them but by doing it you loose compatibility with
clients like Android 2 or other kind of old browsers  still in use... So it
depends on the level of compatibility that we want to achieve to disable or
keep them... Anyway let me check which clients will become incompatible and
I'll be back to you tomorrow.

File: jobs/haproxy/spec (hunk 1 above)
Comment (lcacciagioni) [inline jobs/haproxy/spec]: So with this config we set the base line of support for the following:
* Firefox 1
* Chrome 1
* IE 7
* Opera 5
* Safari 1
* Windows XP IE8
* **Android 2.3**
* **Java 7**
I think that the last 2 are the most important we have to see if we take those 2 out if the support for Android 2.3 & Java 7 is still good.

Comment (lcacciagioni) [inline jobs/haproxy/spec]: For example with a default like: 
`ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256`
we elevate the baseline to:
* Firefox 27
* Chrome 30
* IE 11 on Windows 7
* Edge
* Opera 17
* Safari 9
* Android 5.0
* Java 8
Which is a lot safer but still no java 7 which is quite used specially in enterprise environments.

Comment (lcacciagioni) [inline jobs/haproxy/spec]: Anyway with the cipher suite + no-tls-tickets + HSTS I have achieved an A+ in ssllabs so you tell me if I still have to take those out and I'll do it

**Comment by shinji62** [comment]
Pretty agree on this Cipher.

**Pull request merged**